File: app/src/main/java/com/security/passwordmanager/settings/SettingsRepository.java
package com.security.passwordmanager.settings;

import android.content.Context;

import androidx.annotation.NonNull;

import com.security.passwordmanager.data.MainDatabase;

public class SettingsRepository {

    private final SettingsDao settingsDao;

    public SettingsRepository(@NonNull Context context) {
        settingsDao = MainDatabase.getDatabase(context).settingsDao();
    }

    public void updateTheme(@Support.ThemeDef String theme) {
        Support.Settings settings = new Support.Settings(theme);
        settingsDao.addSettings(settings);
    }

    public @Support.ThemeDef String getTheme() {
        Support.Settings settings = settingsDao.getSettings();
        if (settings == null)
            updateTheme(Support.SYSTEM_THEME);

        return settingsDao.getSettings().getTheme();
    }
}

File: app/src/main/java/com/security/passwordmanager/settings/SettingsRepository.kt
package com.security.passwordmanager.settings

import com.security.passwordmanager.model.Settings
import com.security.passwordmanager.model.Themes

class SettingsRepository(private val settingsDao: SettingsDao) {

    fun updateTheme(email: String, theme: Themes) = settingsDao.updateTheme(email, theme.themeName)

    fun updateUsingBeautifulFont(email: String, usingBeautifulFont: Boolean) =
        settingsDao.updateUsingBeautifulFont(email, usingBeautifulFont)

    fun updateDataHints(email: String, usingDataHints: Boolean) =
        settingsDao.updateDataHints(email, usingDataHints)

    fun updateUsingBottomView(email: String, usingBottomView: Boolean) =
        settingsDao.updateUsingBottomView(email, usingBottomView)

    fun getSettings(email: String) = settingsDao.getSettings(email) ?: Settings()

    suspend fun getSuspendSettings(email: String) =
        settingsDao.getSuspendSettings(email) ?: Settings()
}